pkg/operator/cluster/ceph/mon/mon.go

```diff
@@ -134,61 +107,67 @@ func (c *Cluster) Start() error {
 		return fmt.Errorf("failed to initialize ceph cluster info. %+v", err)
 	}
 
-	if len(c.clusterInfo.Monitors) < c.Size {
-		c.startMons()
-	} else {
-		// Check the health of a previously started cluster
-		if err := c.checkHealth(); err != nil {
-			logger.Warningf("failed to check mon health %+v. %+v", c.clusterInfo.Monitors, err)
-		}
-	}
-
-	return nil
+	return c.startMons()
 }
 
 func (c *Cluster) startMons() error {
-	// init the mons config
-	mons := c.initMonConfig(c.Size)
+	// create statefulset with replicas 0, the loop below does the scale up to the desired c.Size
+	c.startStatefulSet(0)
 
-	// Assign the pods to nodes
-	if err := c.assignMons(mons); err != nil {
-		return fmt.Errorf("failed to assign pods to mons. %+v", err)
-	}
+	mons := []*monConfig{}
 
-	// Start one monitor at a time
+	// Init the mons config
 	for i := len(c.clusterInfo.Monitors); i < c.Size; i++ {
-		// Init the mon IPs
-		if err := c.initMonIPs(mons[0 : i+1]); err != nil {
-			return fmt.Errorf("failed to init mon services. %+v", err)
+		logger.Debugf("mon endpoints used are: %s", c.clusterInfo.MonEndpoints())
+		// this scales up the mons to the desired c.Size one by one through the loop
+		if err := c.updateStatefulSet(int32(i + 1)); err != nil {
+			return fmt.Errorf("failed to scale up replicas for mon statefulset to %d. %+v", i+1, err)
 		}
 
-		// save the mon config after we have "initiated the IPs"
-		if err := c.saveMonConfig(); err != nil {
-			return fmt.Errorf("failed to save mons. %+v", err)
+		monName := getMonNameForID(i)
+
+		// wait until the mon Pod is ready before we get the IP
+		// TODO Maybe change this to only check for Pod is in `Running` state
+		if err := c.waitForPodReady(monName); err != nil {
+			return fmt.Errorf("failed waiting for mon pod %s being ready. %+v", monName, err)
 		}
 
-		// make sure we have the connection info generated so connections can happen
-		if err := WriteConnectionConfig(c.context, c.clusterInfo); err != nil {
-			return err
+		// get mon Pod IP and add it to endpoints
+		// this is needed as the Mon Pod endpoint watcher is only active after
+		// the first full Mon start run
+		podIP, err := c.getMonIP(monName)
+		if err != nil {
+			return fmt.Errorf("failed getting ip from mon pod %s. %+v", monName, err)
+		}
+		c.clusterInfo.MonMutex.Lock()
+		c.clusterInfo.Monitors[monName] = mon.ToCephMon(monName, podIP, mon.DefaultPort)
```

Comment:
in your tests did the podIP change when you killed a mon? When I tried killing a mon in minikube, the same podIP was used in the new pod. I'm anticipating that the mons will not be happy if they come up with a new ip